### PR TITLE
Issue #6615: Add support for Java 14 switch

### LINKED ISSUE #6615: Support for Java 14 switch/yield expression
Running checkstyle validation on a class that contains switch expression:
```
        return switch (c.getLocation().toLowerCase()) {                                                                                                                                                                      
            case "query" -> String.format("ARGS_GET:%s", c.getFullPath().get(0));
            case "header" -> String.format("REQUEST_HEADERS:%s", c.getFullPath().get(0));
            case "cookie" -> String.format("REQUEST_COOKIES:%s", c.getFullPath().get(0));
            default -> "";                                                      
        };                                                                      
```

fails with:
```
unexpected token: switch
```
And also
```
expecting EOF, found 'case'
```

------

FIX Summary:
new Tokens are created. Syntax should be supported, Checks will be updated later on, See follow up issues.  
see https://github.com/checkstyle/checkstyle/pull/8449 for more details

-------

<a href="https://www.bountysource.com/issues/71977640-fails-with-java-12-switch-expression">

</a>

### PR DESCRIPTION
**Pull request by octylFractal**
This includes allowing switch as an expression, i.e. switch expressions, and allowing mutliple expressions in both labels and rules as according to the JLS.

Relevant JLS sections: [14.11 The `switch` Statement](https://docs.oracle.com/javase/specs/jls/se14/html/jls-14.html#jls-14.11), [15.28 `switch` Expressions](https://docs.oracle.com/javase/specs/jls/se14/html/jls-15.html#jls-15.28)

Of note is that I'm not too familiar with how ANTLR 2 works, so I did some "close enough" grammar that accepts invalid Java (allows mixed labels + rules). I'm not sure if this is OK or not.

This fixes #6615.

### CODE REVIEW

**Comment by octylFractal** [comment]
I'm not quite sure why, but this change introduces an oddity with the lambda handling due to my use of the `LAMBDA` token. This no longer parses, and I honestly cannot figure out why. If someone more experienced with writing the grammar would be willing to assist, I would be grateful.
```java
class CollectSpliterators {

    void indexed() {
        IntConsumer consumer = (IntConsumer) i -> {
        };
    }

}
```

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJava14SwitchExpression.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    int signArithmetic(Day day) {
+        return - switch (day) {
+            case MON, FRI, SUN -> 6;
+            case TUE -> 7;
+            case THU, SAT -> 8;
+            case WED -> 9;
+        };
+    }
+
+    int usedOnBothSidesOfAriphmeticExpression(Day day) {
```
Comment (tdaniely) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJava14SwitchExpression.java:116]: typo

Comment (Gunthervg) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJava14SwitchExpression.java:116]: @octylFractal Can you solve the comment?

Comment (Gunthervg) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJava14SwitchExpression.java:116]: @octylFractal I've created a PR which solves the typo: https://github.com/octylFractal/checkstyle/pull/1

@tdaniely If you wish you could use my PR (credits for @octylFractal)

**Comment by romani** [comment]
@nmancus1 , please take a look at this PR, ones you come to switches

**Comment by Gunthervg** [comment]
Has this issue been resolved? Still exception using version 8.35:
"unexpected token: switch"

**Comment by romani** [comment]
I am closing this PR as we have better implementation at #8449

**Pull request closed without merging**